rasa/core/policies/ted_policy.py

```diff
@@ -199,7 +199,7 @@ class TEDPolicy(Policy):
         # If 'True' use value relative embeddings in attention
         VALUE_RELATIVE_ATTENTION: False,
         # Max position for relative embeddings
-        MAX_RELATIVE_POSITION: None,
+        MAX_RELATIVE_POSITION: 5,
```

Comment:
does this mean we can also remove [this check](https://github.com/RasaHQ/rasa/blob/b54f901f140510a8d20aadf8c6f3a415be22ec6a/rasa/utils/tensorflow/transformer.py#L59) ?

Reply:
I'm pretty sure the change is correct but I think this require's some double checking

Reply:
My understanding is that strictly speaking, we only need this value at all if either: `self.use_key_relative_position` or `self.use_value_relative_position` are true (see [here](https://github.com/RasaHQ/rasa/blob/b54f901f140510a8d20aadf8c6f3a415be22ec6a/rasa/utils/tensorflow/transformer.py#L82-L118)).

I was concerned that setting it to a default value (rather than `None`) might mislead people to believe it will always be relevant (and used)... but we're not very disciplined about this in the rest of the code either (e.g. we call `self._create_relative_embeddings()` even though this doesn't apply if neither of the above booleans are true).

That's obviously out of scope for this, all that to say maybe the correct thing to do would be to create another issue around cleaning this up:
- making `MAX_RELATIVE_POSITION` optional, but required when either: `self.use_key_relative_position` or `self.use_value_relative_position`
Moving things like incrementing of `MAX_RELATIVE_POSITION` into parts of the code that can only be reached if the above conditions (one of boolean true and value set) are met.

What do you think? I am happy to create and put on research board.

Reply:
> does this mean we can also remove [this check](https://github.com/RasaHQ/rasa/blob/b54f901f140510a8d20aadf8c6f3a415be22ec6a/rasa/utils/tensorflow/transformer.py#L59) ?

Yes good point, will remove

Reply:
@koernerfelicia there was a discussion in the research channel with on this and we settled that it would be fine to set it to 5 even though this value should not be seen as an absolute endorsement, just like all the other hyperparameters parameters. I don't think an invalid default value makes much sense tbh. If you turn on key or value relative position why should the training crash. But obviously you should think about your hyperparameters.

Reply:
Hmm, just read your comment again, and understand that you also bring up a new aspect here. Users that seeing that `max_relative_position` is set to a specific value and thinking that this is what's happening. I'd be relatively comfortable taking that risk as this config parameter is definitely buried deep either in the code or the documentation. So if people go there, they are definitely more involved/advanced. They might then also see the other two relevant variables.
I think to decrease chance of user failure, I would improve the comment, saying that this is only used if either key or value relative position are turned on. What to you think @koernerfelicia?

Reply:
Yes, your second interpretation is what I was trying to convey (_"Users that seeing that max_relative_position is set to a specific value and thinking that this is what's happening"_). To be clear: I'm not concerned about endorsing 5 as a default value, and I definitely agree that we should not crash. I just wouldn't want the default value being set to suggest to the user that the behaviour is also the default. To that end a comment would definitely work!